modules/installation-configuration-parameters.adoc

```diff
@@ -2603,17 +2603,24 @@ Valid names include:
 --
 endif::ibm-cloud[]
 
-ifdef::vsphere[]
-
+ifdef::agent,vsphere[]
```

Comment:
I'm not sure I understand why you opened this ifdef here and then also on line 2616. I think my question is do you need line 2616? Just checking/learning: is it because you close this opening one down on line 2958?

Reply:
So technically this ifdef and the one on line 2616 are different because they check for slightly different conditions. This ifdef, which is closed on line 2958, basically says "Rule #1: all of the content within these lines can be included if `agent OR vsphere` is true, unless there's an exception".

The ifdef on line 2616 (which closes on line 2622) is basically treated like an exception to that `agent OR vsphere` condition, because it `ONLY` includes content for agent. It's basically saying "here's an exception to rule #1, the content from lines 2616 to 2622 can be included `ONLY` if `agent` is true, but can't be included in vsphere". Let me know if that makes sense, or if you'd like me to explain further.

TLDR the ifdef on line 2606 makes a general rule and the ifdefs like the ones on lines 2616 and 2623 make exceptions to that rule in order to be more specific

Reply:
I'm picking it up now. There just may be hope for me yet 😄 . Merging